engine/runtime/src/main/java/org/enso/interpreter/runtime/builtin/Error.java

```diff
@@ -207,4 +219,135 @@ public Atom makeModuleDoesNotExistError(String name) {
   public Atom makeNotInvokableError(Object target) {
     return notInvokableError.newInstance(target);
   }
+
+  /** Represents plain Java exception as a {@link TruffleObject}.
+   */
+  @ExportLibrary(InteropLibrary.class)
+  static final class WrapPlainException extends AbstractTruffleException {
+    private final AbstractTruffleException prototype;
+    private final Throwable original;
+
+    private WrapPlainException(Throwable cause) {
+      super(cause.getMessage(), cause, AbstractTruffleException.UNLIMITED_STACK_TRACE, null);
+      this.prototype = null;
+      this.original = cause;
+    }
+
+    private WrapPlainException(AbstractTruffleException prototype, Throwable original) {
+      super(prototype);
+      this.prototype = prototype;
+      this.original = original;
+    }
+
+    static AbstractTruffleException wrap(Throwable cause, Context ctx) {
+      var env = ctx.getEnvironment();
+      if (env.isHostException(cause)) {
+        var orig = env.asHostException(cause);
+        return new WrapPlainException((AbstractTruffleException) cause, orig);
+      } else if (cause instanceof AbstractTruffleException truffleEx) {
+        return truffleEx;
+      } else {
+        return new WrapPlainException(cause);
+      }
+    }
+
+    @ExportMessage
+    boolean hasExceptionMessage() {
+      return getMessage() != null;
+    }
+
+    @ExportMessage
+    public Object getExceptionMessage() {
+      return Text.create(getMessage());
+    }
+
+    @ExportMessage
+    String toDisplayString(boolean sideEffects) {
+      return original.toString();
+    }
+
+    @ExportMessage
+    Object getMembers(boolean includeInternal) {
+      return Array.empty();
+    }
+
+    @ExportMessage
+    boolean hasMembers() {
+      return true;
+    }
+
+    @ExportMessage
+    boolean isMemberInvocable(String member, @CachedLibrary(limit="1") InteropLibrary delegate) {
+      if (
+        "has_type".equals(member) ||
+        "getMessage".equals(member)
+      ) {
+        return true;
+      }
+      if (prototype == null) {
+        return false;
+      }
+      return delegate.isMemberInvocable(prototype, member);
```

Comment:
It kinda feels like this would be easier to read as
```suggestion
boolean knownMembers = "has_type".equals(member) || "getMessage".equals(member)
return knownMembers || (prototype != null && delegate.isMemberInvocable(prototype, member);
```

Reply:
+1 to that